src/controllers/developer.ts

```diff
@@ -35,25 +40,29 @@ export const listAllDatasets = async (req: Request, res: Response, next: NextFun
 
 export const displayDatasetPreview = async (req: Request, res: Response) => {
   const dataset = singleLangDataset(res.locals.dataset, req.language);
```

Comment:
this is relying on the `fetchDataset()` middlware
```
developer.get('/:datasetId', fetchDataset(Include.All), displayDatasetPreview);
```
which calls an endpoint which includes a user perms check on the backend.

We might want to either add a bypass for users with the developer role, or it might be more explicit to force the developer frontend to use an entirely different endpoint?